txnsync/txngroups.go

```diff
@@ -60,23 +65,52 @@ func encodeTransactionGroups(inTxnGroups []transactions.SignedTxGroup) ([]byte,
 					stub.BitmaskGroup.SetBit(index)
 				}
 				if err := stub.deconstructSignedTransactions(index, &txn); err != nil {
-					return nil, fmt.Errorf("failed to encodeTransactionGroups: %w", err)
+					return nil, false, fmt.Errorf("failed to encodeTransactionGroups: %w", err)
 				}
 				index++
 			}
 		}
 	}
 	stub.finishDeconstructSignedTransactions()
 
-	return stub.MarshalMsg(protocol.GetEncodingBuf()[:0]), nil
+	encoded := stub.MarshalMsg(protocol.GetEncodingBuf()[:0])
+
+	if len(encoded) > 1000 && (1.0 / compressionSpeed) < (compressionSavings / float32(dataExchangeRate)) {
```

Comment:
1. move the magic number 1000 into it's own constant - `minEncodedTransactionGroupsCompressionThreshold`
2. Refactor the expression `(1.0 / compressionSpeed) < (compressionSavings / float32(dataExchangeRate))` so it would be more understandable.  I'm pretty sure that the expression is wrong because it's not taking into account the `len(encoded)`. More specifically :
projectedUncompressedSendingTime = len(encoded) / dataExchangeRate
projectedCompressedSendingTime = len(encoded) / compressionSpeed + len(encoded)* (1-compressionSavings) / dataExchangeRate

and the if statement should determine if projectedUncompressedSendingTime is more than projectedCompressedSendingTime

Reply:
len(encoded) cancels out on both sides. However, I can still format it to multiply len(encoded) on both sides of the expressions to make the code understandable

Reply:
yes, that's true. I think that few lines that explain how you got to this formula would be a good idea.
also, it can be simplified to `compressionSavings * compressionSpeed > dataExchangeRate`, right ?